portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,27 +44,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     String limitString = request.getParameter("limit");
 
-    // Convert the input to an int.
-    int limit;
+    // Use FetchOptions to limit number of comments
+    FetchOptions fetch;
     try {
-      limit = Integer.parseInt(limitString);
+      fetch = FetchOptions.Builder.withLimit(Integer.parseInt(limitString));
     } catch (NumberFormatException e) {
-      System.err.println("Could not convert to int: " + limitString);
-      limit = -1;
+      fetch = FetchOptions.Builder.withDefaults();
     }
 
+    //Add comments to ArrayList 
```

Comment:
nit: please add a space at the beginning of a comment. Also comments should be sentences, which generally include periods at the end. I don't think this is addressed in your style guide, but is a general best practice.